src/lib/support/Span.h

```diff
@@ -38,6 +41,12 @@ class Span
 
     const T * data() const { return mDataBuf; }
     size_t size() const { return mDataLen; }
+    bool empty() const { return size() == 0; }
+    bool data_equal(const Span & other) const
+    {
+        return (data() != nullptr) && (other.data() != nullptr) && (size() == other.size()) &&
+            (memcmp(data(), other.data(), size() * sizeof(T)) == 0);
```

Comment:
So now if one span is empty with null data ptr and the other empty with non-null data ptr they will test not equal.  Is that the intent?  Is that expected?   I would have expected something like:

```
  return size() == other.size() && (size() == 0 || memcmp(....));
```

Reply:
I believe "0" size should not care about data pointer, anything else should memcmp.

This implementationseems to make "data_equal" always return false if data is nullptr (even if both sides are nullptr) and that seems incorrect for sure.

Reply:
I've updated the implementation. Hopefully, this will be the last round. Thanks for all the comments!